R-packages/modeltools/tests/testthat/test-response.R

```diff
@@ -0,0 +1,42 @@
+library(tibble)
+
+test_that("add_response_columns works for epiweeks", {
+    df <- tibble(
+        geo_value = "oh",
+        time_value = seq.Date(as.Date("2021-01-01"), as.Date("2021-01-31"), "day"),
+        `value-0:my_response` = 1:31
+    )
+    out <- add_response_columns(df, "my_response", as.Date("2021-01-30"), "epiweek", 1:2)
+
+    expect_equal(names(out),
+                 c("geo_value", "time_value", "value-0:my_response",
+                   "value+1:my_response", "value+2:my_response"))
+    expect_equal(out$geo_value, df$geo_value)
+    expect_equal(out$time_value, df$time_value)
+    expect_equal(out$`value-0:my_response`, df$`value-0:my_response`)
+    expect_equal(out$`value+1:my_response`,
+                 c(seq(35, 196, 7), seq(199.5, 217, 3.5), NA))
```

Comment:
No, actually this is the important part of this function.  The forecast date is the important number here.  The forecast date is 2021-01-30, a Saturday, so the +1 epiweek ahead is from Sunday to Saturday, 1 to 7 days ahead.  So this function computes the 1 through 7 day sum from *every* data point rather than the +1 epiweek ahead from each point.  Doing the latter reduces the data set size by a factor of 7 since the epiweek sums are identical for all days in the week.

Reply:
Ok that makes sense. To clarify: If the forecast date was 2021-01-29 (Friday), the +1 epiweek ahead is 2 to 8 days ahead, so this function computes the 2 through 8 day sum for every data point?

Other than that, looks good to me.

Reply:
Yes, exactly.  I can add a test case that makes this more obvious if you would like.